pkg/platform/abstract/project/external/leader/iguazio/types.go

```diff
@@ -8,20 +10,95 @@ type Project struct {
 	Data ProjectData `json:"data,omitempty"`
 }
 
+func CreateProjectFromProjectConfig(projectConfig *platform.ProjectConfig) Project {
+	return Project{
+		Data: ProjectData{
+			Type: ProjectType,
+			Attributes: ProjectAttributes{
+				Name:        projectConfig.Meta.Name,
+				Namespace:   projectConfig.Meta.Namespace,
+				Labels:      labelMapToList(projectConfig.Meta.Labels),
+				Annotations: labelMapToList(projectConfig.Meta.Annotations),
+				Description: projectConfig.Spec.Description,
+			},
+		},
+	}
+}
+
+func (pl *Project) GetConfig() *platform.ProjectConfig {
+	return &platform.ProjectConfig{
+		Meta: platform.ProjectMeta{
+			Name:        pl.Data.Attributes.Name,
+			Namespace:   pl.Data.Attributes.Namespace,
+			Annotations: pl.labelListToMap(pl.Data.Attributes.Annotations),
+			Labels:      pl.labelListToMap(pl.Data.Attributes.Labels),
+		},
+		Spec: platform.ProjectSpec{
+			Description: pl.Data.Attributes.Description,
+		},
+		Status: platform.ProjectStatus{
+			AdminStatus:       pl.Data.Attributes.AdminStatus,
+			OperationalStatus: pl.Data.Attributes.OperationalStatus,
+		},
+	}
+}
+
+func labelMapToList(labelMap map[string]string) []Label {
+	var labelList []Label
+
+	for labelName, labelValue := range labelMap {
+		labelList = append(labelList, Label{Name: labelName, Value: labelValue})
+	}
+
+	return labelList
+}
+
+func (pl *Project) labelListToMap(labelList []Label) map[string]string {
```

Comment:
Why is this a method of Project and labelMapToList is not? they are symmetrical
as generic helpers they shouldn't be in `types.go` but wherever they are used or in some utils/helpers module close to it